runners/maven-plugin/build.gradle.kts

```diff
@@ -1,95 +1,101 @@
+import org.gradle.kotlin.dsl.support.appendReproducibleNewLine
 import org.jetbrains.CrossPlatformExec
-import org.jetbrains.SetupMaven
 import org.jetbrains.registerDokkaArtifactPublication
 
 plugins {
     org.jetbrains.conventions.`kotlin-jvm`
     org.jetbrains.conventions.`maven-publish`
+    org.jetbrains.conventions.`maven-cli-setup`
 }
 
-val setupMaven by tasks.register<SetupMaven>("setupMaven")
-
 dependencies {
     implementation(project(":core"))
-    implementation("org.apache.maven:maven-core:${setupMaven.mavenVersion}")
-    implementation("org.apache.maven:maven-plugin-api:${setupMaven.mavenVersion}")
-    implementation("org.apache.maven.plugin-tools:maven-plugin-annotations:${setupMaven.mavenPluginToolsVersion}")
+    implementation("org.apache.maven:maven-core:${setupMavenProperties.mavenVersion}")
+    implementation("org.apache.maven:maven-plugin-api:${setupMavenProperties.mavenVersion}")
+    implementation("org.apache.maven.plugin-tools:maven-plugin-annotations:${setupMavenProperties.mavenPluginToolsVersion}")
     implementation("org.apache.maven:maven-archiver:2.5")
     implementation(kotlin("stdlib-jdk8"))
 }
 
-val mavenBuildDir = setupMaven.mavenBuildDir
-val mavenBinDir = setupMaven.mavenBinDir
-
-tasks.clean {
-    delete(mavenBuildDir)
-    delete(mavenBinDir)
-}
+val mavenPluginTaskGroup = "maven plugin"
 
-val generatePom by tasks.registering(Copy::class) {
+val generatePom by tasks.registering(Sync::class) {
     description = "Generate pom.xml for Maven Plugin Plugin"
+    group = mavenPluginTaskGroup
 
     val dokka_version: String by project
     inputs.property("dokka_version", dokka_version)
 
-    from("$projectDir/pom.tpl.xml") {
-        rename("(.*).tpl.xml", "$1.xml")
-    }
-    into(setupMaven.mavenBuildDir)
-
-    eachFile {
-        filter { line ->
-            line.replace("<maven.version></maven.version>", "<maven.version>${setupMaven.mavenVersion}</maven.version>")
-        }
-        filter { line ->
-            line.replace("<version>dokka_version</version>", "<version>$dokka_version</version>")
-        }
-        filter { line ->
-            line.replace(
-                "<version>maven-plugin-plugin</version>",
-                "<version>${setupMaven.mavenPluginToolsVersion}</version>"
-            )
-        }
+    val pomTemplateFile = layout.projectDirectory.file("pom.tpl.xml")
+
+    from(pomTemplateFile) {
+        rename { it.replace(".tpl.xml", ".xml") }
+
+        expand(
+            "mavenVersion" to setupMavenProperties.mavenVersion,
+            "dokka_version" to dokka_version,
+            "mavenPluginToolsVersion" to setupMavenProperties.mavenPluginToolsVersion,
+        )
     }
+
+    into(temporaryDir)
```

Comment:
Instead of worrying about Sync/Copy tasks overwriting, and having to use `presever {}`, I think it's best to always use Sync into the tasks temporary dir, and use final Sync task (`prepareMavenPluginBuildDir`) to sync all the files in one go.